src/object/cli_shard.c

```diff
@@ -925,7 +925,9 @@ dc_rw_cb(tse_task_t *task, void *arg)
 				int rc1;
 
 				rc1 = dc_shard_update_size(rw_args, rc);
-				D_ASSERT(rc1 == 0);
+				D_ASSERT(rc1 == 0 || rc1 == -DER_FETCH_AGAIN);
```

Comment:
this change looks incorrect for some cases, it is in the case of singv overwrite, seems need some other code change, I'll check more details later and will let you know. thx

Reply:
I pushed https://github.com/daos-stack/daos/pull/7350 with some refines and more test case added